mesonbuild/utils/universal.py

```diff
@@ -756,40 +757,54 @@ def windows_detect_native_arch() -> str:
             raise EnvironmentException('Unable to detect native OS architecture')
     return arch
 
-def detect_vcs(source_dir: T.Union[str, Path]) -> T.Optional[T.Dict[str, str]]:
+@dataclasses.dataclass
+class VcsData:
+    name: str
+    cmd: str
+    repo_dir: str
+    get_rev: 'T.List[str]'
+    rev_regex: str
+    dep: str
+    wc_dir: 'T.Optional[str]'
+
+def detect_vcs(source_dir: T.Union[str, Path]) -> 'VcsData':
     vcs_systems = [
-        {
-            'name': 'git',
-            'cmd': 'git',
-            'repo_dir': '.git',
-            'get_rev': 'git describe --dirty=+ --always',
-            'rev_regex': '(.*)',
-            'dep': '.git/logs/HEAD'
-        },
-        {
-            'name': 'mercurial',
-            'cmd': 'hg',
-            'repo_dir': '.hg',
-            'get_rev': 'hg id -i',
-            'rev_regex': '(.*)',
-            'dep': '.hg/dirstate'
-        },
-        {
-            'name': 'subversion',
-            'cmd': 'svn',
-            'repo_dir': '.svn',
-            'get_rev': 'svn info',
-            'rev_regex': 'Revision: (.*)',
-            'dep': '.svn/wc.db'
-        },
-        {
-            'name': 'bazaar',
-            'cmd': 'bzr',
-            'repo_dir': '.bzr',
-            'get_rev': 'bzr revno',
-            'rev_regex': '(.*)',
-            'dep': '.bzr'
-        },
+        VcsData(
+            name = 'git',
+            cmd = 'git',
+            repo_dir = '.git',
+            get_rev = ['git', 'describe', '--dirty=+', '--always'],
+            rev_regex = '(.*)',
+            dep = '.git/logs/HEAD',
+            wc_dir = None,
```

Comment:
There shouldn't be a space around the `=` here.

Reply:
Originally there wasn't. Lint complained. I added it in. Now it is quiet.